caffe2/core/common_gpu.h

```diff
@@ -25,6 +25,8 @@
 #include "caffe2/core/common.h"
 #include "caffe2/core/logging.h"
 
+#include "c10/cuda/math_compat.h"
```

Comment:
common_gpu.h is included in a lot of .cc files that are not built with nvcc. Is it fine to have __host__ annotations in them?

Reply:
It's actually not good (although it actually works) - gcc doesn't allow it, but cuda headers define both `__host__` and `__device__` to empty macro if `__CUDACC__` is not defined (and I happen to have included my compat file after some cuda includes). I'm going to change it.